test/validation/api/ipsec/ipsec.c

```diff
@@ -831,8 +831,6 @@ void ipsec_check_in_one(const ipsec_test_part *part, odp_ipsec_sa_t sa)
 			CU_ASSERT_EQUAL(0, odp_ipsec_result(&result, pkto[i]));
 			CU_ASSERT_EQUAL(part->out[i].status.error.all,
 					result.status.error.all);
-			CU_ASSERT(!result.status.error.all ==
-				  !odp_packet_has_error(pkto[i]));
```

Comment:
This fixes the problem that the validation test checks that odp_packet_has_error() reports an error when odp_ipsec_result() reports an error. But in case odp_ipsec_result() reports success, maybe the validation check should still check that also odp_packet_has_error() reports no-error. So instead of removing the check completely, maybe it should be made conditional on IPsec processing success?

Reply:
Agreed. Updated V4 such that odp_packet_has_error() is checked conditionally when IPsec processing is a success.
Additionally on linux-generic, fixed adjusting the bit fields while removing ipsec_err bit from _odp_packet_flags_t.